eclipse-ankaios/config/startupState.yaml

```diff
@@ -6,7 +6,7 @@ workloads:
       - key: project
         value: chariott
     runtimeConfig: |
-      image: sdvblueprint.azurecr.io/sdvblueprint/eclipse-chariott/service-discovery:0.1.1
+      image: ghcr.io/ladatz/sdvblueprint/eclipse-chariott/service-discovery:0.2.0
```

Comment:
To me it seems that they're a more favorable approaches instead of pulling from a personal GitHub container registry. Especially, since the pulled content/code is already from an Eclipse open source project. Would it be an alternative to directly pull from the Eclipse Chariott repository instead? @ladatz

Reply:
This is a good point, however we are a little blocked by our permissions. We do not want to keep these images in my personal GitHub container registry long-term. We also do not want to host them in the individual projects' Container Registries for now either, because they have blueprint-specific settings and configurations to integrate with the other projects. Using the personal registry was an intermediate step to unblock our testing and development. 

We would like to host them in the eclipse-sdv-blueprints container registry (like what FMS does), but since we are not committers (and do not have write access), I get permission_denied when I try to push Images there.

I can see a few options on how to proceed: 

1. We can have someone with write access to the container registry pull down those images and push them into the blueprints container registry and then update these references to point to those images.
2. We can use this personal GitHub container registry for testing and development for now, merge this PR, and then once we can become committers to the project I can push them to the blueprint container registry myself.

I am open to any other discussion/ suggestions

Reply:
A couple more comments: these are public images in a public container registry, but it is listed under my personal organization. I chose this container registry because it is most similar to the way that we have to do PRs without committer permissions: we have to create a fork in our personal organization.

Reply:
I would prefer that we become write access by becoming official committers. Then you can tag the images again with something like `docker image tag <image_id> ghcr.io/eclipse-sdv-blueprints/software-orchestration/eclipse-chariott/service-discovery:0.2.0`and push it to the package repository. All the temporary access would work of course somehow, but we shall also add some automatic release build later which is basically a container image published to the package repository as well.

Reply:
Ok, I see where this is coming from.

In general I would prefer it if the build process for the containers is fully transparent and it is comprehensible for an outsider what actually has been put into the containers.
Because of that my proposal is that you push the Dockerfile including the configuration for the build to the blueprint repository. You can then create a GitHub action for building and then pushing the container to the GHCR of this blueprint. This way no developer needs to locally build or manually push the container somewhere and it is clear what has been built.

This is also the approach we took for the containers in the FMS-Blueprint: https://github.com/eclipse-sdv-blueprints/fleet-management/blob/main/.github/workflows/containerize_fms_forwarder.yaml . So you could use these actions as a template.

I cannot give you specific rights to the repository. The rights are managed by the Eclipse Foundation and the only way to get some kind of writing right to the repository is by being a commiter.

Reply:
Thanks for the suggestion. I will look at our options (since our use case is slightly different than the FMS case, the code is hosted in other repos) to try to unblock ourselves.

Reply:
I am still blocked by permissions when trying to run the worflow that I am working on from my fork (the only way I can test at the moment). The workflow worked to push images to "my" ladatz GitHub container registry, but not to the eclipse-sdv-blueprints registry. I still get a 403.

Reply:
My guess is that it is intended to only allow pushes from within the context of the eclipse-sdv-blueprints repository and registry. My hope would be that the same workflow works when you contribute and run it in this repository. Would you be willing to contribute the workflow?

Reply:
Thanks @eriksven PR here: https://github.com/eclipse-sdv-blueprints/software-orchestration/pull/15

Reply:
Resolving since the images from the blueprint container registry are used now